packages/custom-url/src/Infrastructure/Doctrine/EventListener/CustomUrlRouteHistoryListener.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\CustomUrl\Infrastructure\Doctrine\EventListener;
+
+use Doctrine\ORM\Event\PostFlushEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Sulu\CustomUrl\Domain\Model\CustomUrl;
+use Sulu\CustomUrl\Domain\Model\CustomUrlRoute;
+
+/**
+ * Manages history routes when custom URL paths change.
+ * Marks old routes as history and links them to new routes.
+ */
+class CustomUrlRouteHistoryListener
```

Comment:
```suggestion
/**
 * @internal No BC promises are given for this class. It may be changed or removed at any time.
 *
 * Manages history routes when custom URL paths change.
 * Marks old routes as history and links them to new routes.
 */
class CustomUrlRouteHistoryListener
```